src/cogent3/core/annotation_db.py

```diff
@@ -1275,18 +1275,19 @@ def add_records(self, reduced: dict) -> None:
         # Can we really trust text case of "ID" and "Parent" to be consistent
         # across sources of gff?? I doubt it, so more robust regex likely
         # required
-        rows = []
-        for record in reduced.values():
+        keys = tuple(reduced)
+        for key in keys:
             # our Feature code assumes start always < stop,
             # we record direction using Strand
+            record = reduced[key]
             spans = numpy.array(sorted(record["spans"]), dtype=int)  # sorts the rows
             spans.sort(axis=1)
             record["start"] = int(spans.min())
             record["stop"] = int(spans.max())
             record["spans"] = spans
-            rows.append(tuple(record.get(c) for c in col_order))
+            self.db.execute(sql, tuple(record.get(c) for c in col_order))
```

Comment:
lgtm. Though, I wonder about the performance implications of many individual inserts rather than one large batched insert when there are many rows. Potentially could have some kind of batch size, though likely overcomplicating it for now.

Reply:
True, if we wind up doing batches of lines, this change will not be worth it. I'll revert that.